plugins/usi-dock/fu-usi-dock-dmc-device.c

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2021 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-usi-dock-dmc-device.h"
+
+struct _FuUsiDockDmcDevice {
+	FuUsbDevice parent_instance;
+};
+
+G_DEFINE_TYPE(FuUsiDockDmcDevice, fu_usi_dock_dmc_device, FU_TYPE_USB_DEVICE)
+
+static void
+fu_usi_dock_dmc_device_parent_notify_cb(FuDevice *device, GParamSpec *pspec, gpointer user_data)
+{
+	FuDevice *parent = fu_device_get_parent(device);
+	if (parent != NULL) {
```

Comment:
Seems like this should be a critical failure no?

Reply:
@hughsie could you help to explain this?

Reply:
Ohh I don't think it has to be critical; in the future we might have a parent-less DMC device.